tests/integration/vlan_test.py

```diff
@@ -234,3 +236,87 @@ def create_two_vlans_state():
+            {
+                Interface.NAME: vlana_name,
+                Interface.TYPE: InterfaceType.VLAN,
+                Interface.STATE: InterfaceState.UP,
+                Interface.MTU: 1280,
+                Interface.MAC: "d4:ee:07:25:42:5a",
+                VLAN.CONFIG_SUBTREE: {VLAN.ID: 201, VLAN.BASE_IFACE: "eth1"},
+                Interface.IPV4: {
+                    InterfaceIPv4.ENABLED: True,
+                    InterfaceIPv4.ADDRESS: [
+                        {
+                            InterfaceIPv4.ADDRESS_IP: "192.0.2.251",
+                            InterfaceIPv4.ADDRESS_PREFIX_LENGTH: 24,
+                        }
+                    ],
+                },
+                Interface.IPV6: {
+                    InterfaceIPv6.ENABLED: True,
+                    InterfaceIPv6.ADDRESS: [
+                        {
+                            InterfaceIPv6.ADDRESS_IP: "2001:db8:1::1",
+                            InterfaceIPv6.ADDRESS_PREFIX_LENGTH: 64,
+                        }
+                    ],
+                },
+            },
+            {
+                Interface.NAME: vlanb_name,
+                Interface.TYPE: InterfaceType.VLAN,
+                Interface.STATE: InterfaceState.UP,
+                Interface.MTU: 1280,
+                VLAN.CONFIG_SUBTREE: {VLAN.ID: 202, VLAN.BASE_IFACE: "eth1"},
+                Interface.IPV4: {
+                    InterfaceIPv4.ENABLED: True,
+                    InterfaceIPv4.DHCP: True,
+                    InterfaceIPv4.AUTO_DNS: False,
+                    InterfaceIPv4.AUTO_GATEWAY: False,
+                    InterfaceIPv4.AUTO_ROUTES: False,
+                    InterfaceIPv4.AUTO_ROUTE_TABLE_ID: 100,
+                },
+                Interface.IPV6: {
+                    InterfaceIPv6.ENABLED: True,
+                    InterfaceIPv6.DHCP: True,
+                    InterfaceIPv6.AUTOCONF: True,
+                    InterfaceIPv6.AUTO_DNS: True,
+                    InterfaceIPv6.AUTO_GATEWAY: True,
+                    InterfaceIPv6.AUTO_ROUTES: True,
+                    InterfaceIPv6.AUTO_ROUTE_TABLE_ID: 100,
+                },
+            },
+        ]
+    }
+
+    libnmstate.apply(desired_state)
+    base_state = libnmstate.show()
+    for iface in base_state[Interface.KEY]:
+        if iface[Interface.NAME] == vlana_name:
+            iface[VLAN.CONFIG_SUBTREE][VLAN.ID] = 1001
+        elif iface[Interface.NAME] == vlanb_name:
+            iface[VLAN.CONFIG_SUBTREE][VLAN.ID] = 1002
+
+    desired_state = {
+        Interface.KEY: [
+            {
+                Interface.NAME: vlana_name,
+                VLAN.CONFIG_SUBTREE: {VLAN.ID: 1001},
+            },
+            {
+                Interface.NAME: vlanb_name,
+                VLAN.CONFIG_SUBTREE: {VLAN.ID: 1002},
+            },
+        ]
+    }
+    libnmstate.apply(desired_state)
+    new_state = libnmstate.show()
+    assert base_state == new_state
```

Comment:
Instead of this, you can do:
```
assertlib.assert_state(vlan_on_eth1)
```

Reply:
Instead of verify the vlan interface only, I meant to verify the whole state, in case it impacts the others. As this ever happened in certain old versions.

Reply:
But sometimes routes, DNS from DHCP do have a chance to vary automatically...
At least I can use assertlib.assert_state(base_state) to verify all `interfaces`. What do you think?